### PR TITLE
Hard computer player

### PR DESCRIPTION
**Pull request by gilroman**

### CODE REVIEW

File: src/tic_tac_toe_pair/ai.clj
Comment (pumpkincouture) [inline src/tic_tac_toe_pair/ai.clj]: This should be 'lose' instead of 'loose'

File: src/tic_tac_toe_pair/console.clj
Comment (pumpkincouture) [inline src/tic_tac_toe_pair/console.clj]: Similarly, this should be 'losing'

File: src/tic_tac_toe_pair/ai.clj
Comment (gilroman) [reply src/tic_tac_toe_pair/ai.clj]: Thanks for catching this @pumpkincouture

File: src/tic_tac_toe_pair/ai.clj
Comment (ssunday) [inline src/tic_tac_toe_pair/ai.clj]: This should be indented in

File: src/tic_tac_toe_pair/game.clj
Comment (ssunday) [inline src/tic_tac_toe_pair/game.clj:50]: How would you refactor this so that either player could be computer or AI?

Comment (gilroman) [reply src/tic_tac_toe_pair/game.clj:50]: Thanks for reviewing my PR Sarah!!!
This is the way I would refactor the function. What do you think? 

1. Change the game map so that players can be given a type value of :computer or :human like this:

```
{:player-1 {:token :o :type :computer}
 :player-2 {:token :x :type :human}
 :board [nil nil nil nil nil nil nil nil]
 :current-player :player-1
 }
```

2. Refactor the function so that it calls the correct function depending on the player type.

```
(defn move [game]
  (let [{{current-player-type :type} (:current-player game)} game] 
    (cond 
      (= :computer current-player-type) (get-ai-move)
      (= :human current-player-type) (get-move-location))))
```

Having players be either ai or human is not part of this story, so I will save this work on a separate branch for future use.

File: src/tic_tac_toe_pair/ai.clj
Comment (gilroman) [reply src/tic_tac_toe_pair/ai.clj]: Thanks!!!

File: src/tic_tac_toe_pair/game.clj
Comment (ssunday) [reply src/tic_tac_toe_pair/game.clj:50]: That's like what I was thinking! I like giving the player both a marker and a type

Comment (gilroman) [reply src/tic_tac_toe_pair/game.clj:50]: Great! I definitely plan on refactoring and implementing this.

File: src/tic_tac_toe_pair/rules.clj
Comment (ssunday) [inline src/tic_tac_toe_pair/rules.clj]: Could this be reduced to just being the boolean conditional? Do you need to return true/false like that?

Comment (gilroman) [reply src/tic_tac_toe_pair/rules.clj]: Yes!!!! Great catch. I'll change it to:

`(defn is-game-over? [board]
  (or (is-full? board) (is-game-won? board)))`

Comment (ssunday) [reply src/tic_tac_toe_pair/rules.clj]: That looks perfect

Comment (gilroman) [reply src/tic_tac_toe_pair/rules.clj]: Sarah, thank you for your time looking at this. Do you approve this PR? 
I've been guided to make sure to get explicit approval before merging it.

**Review by pumpkincouture — APPROVED** [review]

File: src/tic_tac_toe_pair/rules.clj
Comment (ssunday) [reply src/tic_tac_toe_pair/rules.clj]: @gilroman Oh my gosh, I meant to hit approve but then I got lost in other stuff. Sorry!!!!!!

Comment (gilroman) [reply src/tic_tac_toe_pair/rules.clj]: @ssunday No worries at all!!! It was great getting your feedback.

**Pull request merged**
